hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/ReadOnlyController.java

```diff
@@ -105,6 +105,11 @@ private void internalReadOnlyGuard() throws IOException {
     }
   }
 
+  private boolean
+    isOperationOnNonMetaTable(final ObserverContext<? extends RegionCoprocessorEnvironment> c) {
```

Comment:
nit: Simply call it "isOnMeta" and return the check c.getEnvironment().getRegionInfo().getTable().equals(TableName.META_TABLE_NAME), moving the negation check to the callers? eg:
...
    if (!isOnMeta(c)) {
      internalReadOnlyGuard();
    } 
...